File: solutions/Binary-Tree-Level-Order-Traversal/Binary-Tree-Level-Order-Traversal.py

‘’’
 Definition for a  binary tree node
 class TreeNode:
     def __init__(self, x):
         self.val = x
         self.left = None
         self.right = None
‘’’
class Solution:
    # @param root, a tree node
    # @return a list of lists of integers
    def levelOrder(self, root):
        ans=[]
        q=[]
        if root==None:
            return ans
        vec=[]
        vec.append(root)
        q.append(vec)
        while len(q):
            vec=q.pop()
            temp=[]
            temp2=[]
            for i in range(len(vec)):
                temp.append(vec[i].val)
                if vec[i].left!=None:
                    temp2.append(vec[i].left)
                if vec[i].right!=None:
                    temp2.append(vec[i].right)
            if len(temp2):
                q.append(temp2)
            if len(temp):
                ans.append(temp)
        return ans
 

File: solutions/Binary-Tree-Level-Order-Traversal/Binary-Tree-Level-Order-Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  const ans = [];
  if (root === null) {
    return ans;
  }
  let levels = [root];
  while (levels.length > 0) {
    const temp = [];
    const curLevel = [];
    while (levels.length > 0) {
      const current = levels.shift();
      curLevel.push(current.val);
      if (current.left !== null) {
        temp.push(current.left);
      }
      if (current.right !== null) {
        temp.push(current.right);
      }
    }
    ans.push(curLevel);
    levels = temp;
  }
  return ans;
};